test/lit/passes/opt_flatten.wast

```diff
@@ -0,0 +1,43 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; Optimize the code, flatten, and run a pass that will verify it is properly
+;; flat.
+;; RUN: wasm-opt %s -all -O1 --flatten --rereloop -S -o - | filecheck %s
```

Comment:
What does the -O1 do for the test? Would it be possible to start out directly at `--flatten`?

Reply:
The -O1 is necessary. I did not investigate in depth why, but I suspect it is something we add during binary reading that opts remove.